project-koku/koku
# COST-3050 Repeated update of timestamp for OCI source

## Jira Ticket

[COST-3050](https://issues.redhat.com/browse/COST-3050)

## Description

This change will add skipping creating manifest records for months with no reports and  a function remove duplicate manifests to prevent repeatedly updating `last_manifest_complete_date` field.

## Testing

1. Checkout Branch
2. Restart Koku
3. Load some oci data with `make create-test-customer` and `make load-test-customer-data test_source=oci`
4. Hit the sources endpoint to get the oci source id, http://127.0.0.1:8000/api/cost-management/v1/sources/
5. Hit endpoint  `http://127.0.0.1:8000/api/cost-management/v1/sources/<id>/stats/`
    1. You should see unique timestamps in `last_manifest_complete_date` field for each manifest. 
    
6. Do more things...

## Notes

Actual endpoint should be similar to this http://127.0.0.1:8000/api/cost-management/v1/sources/11/stats/
